cli/generate.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2022 Nethermind
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cli
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/NethermindEth/sedge/cli/actions"
+	"github.com/NethermindEth/sedge/configs"
+	"github.com/NethermindEth/sedge/internal/pkg/clients"
+	"github.com/NethermindEth/sedge/internal/pkg/generate"
+	"github.com/NethermindEth/sedge/internal/ui"
+	"github.com/NethermindEth/sedge/internal/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// Global vars
+var (
+	generationPath string
+	network        string
+	logging        string
+)
+
+// GenCmdFlags is a struct that holds the flags of the generate command
+type GenCmdFlags struct {
+	CustomFlags
+	executionName     string
+	consensusName     string
+	validatorName     string
+	checkpointSyncUrl string
+	feeRecipient      string
+	noMev             bool
+	mevImage          string
+	mevBoostOnVal     bool
+	noValidator       bool
+	jwtPath           string
+	graffiti          string
+	mapAllPorts       bool
+	fallbackEL        *[]string
+	elExtraFlags      *[]string
+	clExtraFlags      *[]string
+	vlExtraFlags      *[]string
+	relayURL          string
+	mevBoostUrl       string
+	executionApiUrl   string
+	executionAuthUrl  string
+	consensusApiUrl   string
+	waitEpoch         int
+}
+
+func GenerateCmd(sedgeAction actions.SedgeActions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generate [flags]",
+		Short: "Generate new setups according to selected options",
+		Long: `Generate new setups according to selected options and flags.
+
+It will create a 'docker-compose.yml' and a '.env', which you will need later to run the docker-compose script. You can use 'sedge run' to run the script and start the setup.
+
+You can generate:
+- Full Node (execution + consensus + validator)
+- Full Node without Validator (execution + consensus)
+- Execution Node
+- Consensus Node
+- Validator Node
+- Mev-Boost Instance
+`,
+		Args: cobra.NoArgs,
+	}
+
+	cmd.AddCommand(FullNodeSubCmd(sedgeAction))
+	cmd.AddCommand(ExecutionSubCmd(sedgeAction))
+	cmd.AddCommand(ConsensusSubCmd(sedgeAction))
+	cmd.AddCommand(ValidatorSubCmd(sedgeAction))
+	cmd.AddCommand(MevBoostSubCmd(sedgeAction))
+
+	cmd.PersistentFlags().StringVarP(&generationPath, "path", "p", configs.DefaultDockerComposeScriptsPath, "docker-compose scripts generation path")
```

Comment:
`docker-compose-scripts` is still being used for the sedge data dir. This wasn't updated with https://github.com/NethermindEth/sedge/pull/159 bc that PR updated the `DefaultSedgeDataFolderName` message in `configs/messages.go` and not the`DefaultDockerComposeScriptsPath` in `configs/paths.go`. Please fix this in another PR and also fix the use of `./` in the path as you had to do in that PR.

Reply:
Ok, but use the one from `configs/paths.go` now